Statement.php

```diff
@@ -147,17 +156,21 @@ public function errorInfo() {
 
     /**
      * {@inheritdoc}
+     *
+     * @SuppressWarnings("PHPMD.StaticAccess")
      */
     public function execute($params = null) {
-        $rawRequest = $this->mysqlToRequest->transform($this->query, $this->params);
-        $request    = $this->authStrategy->transformRequest($rawRequest);
-        $restClient = $this->restClientFactory->createOne($request->getCurlOptions());
+        $transformedQuery = SqlQuery::quoteUrl(SqlQuery::setParams($this->query, $params !== null ? $params : $this->params));
+        $rawRequest       = $this->mysqlToRequest->transform($transformedQuery);
+        $request          = $this->authStrategy->transformRequest($rawRequest);
+
+        $response = $this->restClient->send($request);
 
-        $method     = strtolower($request->getMethod());
-        $response   = $method === HttpMethods::GET || $method === HttpMethods::DELETE ? $restClient->$method($request->getUrlAndQuery()) : $restClient->$method($request->getUrlAndQuery(), $request->getPayload());
-        $statusCode = $response->getStatusCode();
+        $this->result = Result::create($transformedQuery, $this->formatter->decode($response->getContent()));
+        $this->id     = !empty($this->result['id']) ? $this->result['id'] : null;
+        krsort($this->result);
 
-        return $statusCode === 200 || ($method === HttpMethods::DELETE && $statusCode === 204) ? $this->onSuccess($response, $method) : $this->onError($request, $response);
+        return true;
```

Comment:
is die denn von außen catchbar? weil wenn nein, is das fatal :D, weil du ja die ganze applikation dann schmieren lässt oder wird das dann von sf auch gecatched immer? (Es muss ja iwie die möglichkeit geben, doctrine den error zu signalisieren in meinen augen und das wäre doch das return false, klar mehr kontext wär dann schon gut, aber ich würde da im framework bleiben, wenn das geht...)